### PR TITLE
Holy Paladin: Add a Crusader's Might module that shows the actual CDR

### PR DESCRIPTION
**Pull request by MartijnHols**
TODO: show percentage instead (e.g. if your Holy Shock was on CD for 200 sec (after reductions) and you had a 60 sec CDR, it would show a 23% effective CDR. This is easier to grasp the value of.

# Depends on #466

### CODE REVIEW

File: src/common/format.js [hunk 1]
```diff
@@ -23,12 +23,16 @@ export function formatNumber(number) {
   return formatThousands(number);
 }
 
+export function formatDecimal(number, precision = 2) {
+  return (Math.round((number || 0) * 100) / 100).toFixed(precision);
+}
```
Comment (kfinch) [inline src/common/format.js]: How is this meaningfully distinct from just doing `number.toFixed(precision);` ?

File: src/Parser/HolyPaladin/Modules/Talents/CrusadersMight.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+class CrusadersMight extends Module {
+  static dependencies = {
+    combatants: Combatants,
+    spellUsable: SpellUsable,
+  };
+
+  effectiveHolyShockReductionMs = 0;
+  wastedHolyShockReductionMs = 0;
+  effectiveLightOfDawnReductionMs = 0;
+  wastedLightOfDawnReductionMs = 0;
```
[the remark this answers is not in the document]
Comment (MartijnHols) [reply src/Parser/HolyPaladin/Modules/Talents/CrusadersMight.js]: I don't have a use for it *yet*.

File: src/Parser/HolyPaladin/Modules/Talents/CrusadersMight.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    const holyShockisOnCooldown = this.spellUsable.isOnCooldown(SPELLS.HOLY_SHOCK_CAST.id);
+    if (holyShockisOnCooldown) {
+      const reductionMs = this.spellUsable.reduceCooldown(SPELLS.HOLY_SHOCK_CAST.id, COOLDOWN_REDUCTION_MS);
+      this.effectiveHolyShockReductionMs += reductionMs;
+      this.wastedHolyShockReductionMs += COOLDOWN_REDUCTION_MS - reductionMs;
+    } else {
+      const wastedReductionMs = COOLDOWN_REDUCTION_MS;
+      this.wastedHolyShockReductionMs += wastedReductionMs;
```
[the remark this answers is not in the document]
Comment (MartijnHols) [reply src/Parser/HolyPaladin/Modules/Talents/CrusadersMight.js]: Thanks, I initially did it this way to be consistent with the block on the other end of the else, but I guess it doesn't really help readability.

File: src/common/format.js (hunk 1 above)
Comment (MartijnHols) [reply src/common/format.js]: It isn't, removed it.

**Comment by kfinch** [comment]
guess I should merge this too, huh

**Pull request merged**